In [2]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/housing-dataset/Housing.csv


In [3]:
House_Data = pd.read_csv("/kaggle/input/housing-dataset/Housing.csv")
House_Data.head()

price  area  bedrooms  bathrooms  stories mainroad guestroom basement  \
0  13300000  7420         4          2        3      yes        no       no   
1  12250000  8960         4          4        4      yes        no       no   
2  12250000  9960         3          2        2      yes        no      yes   
3  12215000  7500         4          2        2      yes        no      yes   
4  11410000  7420         4          1        2      yes       yes      yes   

  hotwaterheating airconditioning  parking prefarea furnishingstatus  
0              no             yes        2      yes        furnished  
1              no             yes        3       no        furnished  
2              no              no        2      yes   semi-furnished  
3              no             yes        3      yes        furnished  
4              no             yes        2       no        furnished

In [6]:
#Checking the length of the data frame
House_Data_length = House_Data.sample(n=len(House_Data), random_state=1)
House_Data_length


price   area  bedrooms  bathrooms  stories mainroad guestroom basement  \
62   7070000   6240         4          2        2      yes        no       no   
247  4550000   8400         4          1        4      yes        no       no   
142  5600000  10500         4          2        2      yes        no       no   
107  6125000   6420         3          1        3      yes        no      yes   
483  2940000   6615         3          1        2      yes        no       no   
..       ...    ...       ...        ...      ...      ...       ...      ...   
129  5873000  11460         3          1        3      yes        no       no   
144  5600000   4700         4          1        2      yes       yes      yes   
72   6720000   5020         3          1        4      yes        no       no   
235  4620000   5680         3          1        2      yes       yes       no   
37   7980000   9000         4          2        4      yes        no       no   

    hotwaterheating airconditioning  parking prefarea furnishingstatus  
62               no             yes        1       no        furnished  
247              no              no        3       no      unfurnished  
142              no              no        1       no   semi-furnished  
107              no              no        0      yes      unfurnished  
483              no              no        0       no   semi-furnished  
..              ...             ...      ...      ...              ...  
129              no              no        2      yes   semi-furnished  
144              no             yes        1       no        furnished  
72               no             yes        0      yes      unfurnished  
235              no             yes        1       no   semi-furnished  
37               no             yes        2       no        furnished  

[545 rows x 13 columns]

In [4]:
House_Data.shape

(545, 13)

* **This indicates the House dataset has 545 rows and 13 columns.**

In [5]:
#checking mainroad proximity
House_Data['mainroad'].value_counts()

mainroad
yes    468
no      77
Name: count, dtype: int64

In [10]:
pd.get_dummies(House_Data['airconditioning']).astype(int).head()


no  yes
0   0    1
1   0    1
2   1    0
3   0    1
4   0    1

In [23]:
House_Data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 545 entries, 0 to 544
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   price             545 non-null    int64 
 1   area              545 non-null    int64 
 2   bedrooms          545 non-null    int64 
 3   bathrooms         545 non-null    int64 
 4   stories           545 non-null    int64 
 5   mainroad          545 non-null    object
 6   guestroom         545 non-null    object
 7   basement          545 non-null    object
 8   hotwaterheating   545 non-null    object
 9   airconditioning   545 non-null    object
 10  parking           545 non-null    int64 
 11  prefarea          545 non-null    object
 12  furnishingstatus  545 non-null    object
dtypes: int64(6), object(7)
memory usage: 55.5+ KB


In [ ]:
House_Data.describe()

from sklearn.metrics import mean_squared_error, mean_absolute_error

# Calculate Mean Squared Error (MSE) and Mean Absolute Error (MAE) for training data
Y_pred_train = model.predict(X_train)
mse_train = mean_squared_error(Y_train, Y_pred_train)
mae_train = mean_absolute_error(Y_train, Y_pred_train)
print("Training MSE:", mse_train)
print("Training MAE:", mae_train)

# Calculate Mean Squared Error (MSE) and Mean Absolute Error (MAE) for testing data
mse_test = mean_squared_error(Y_test, Y_pred)
mae_test = mean_absolute_error(Y_test, Y_pred)
print("Testing MSE:", mse_test)
print("Testing MAE:", mae_test)


In [ ]:
import pandas as pd

# Load the data from the CSV file
House_Data = pd.read_csv("/kaggle/input/housing-dataset/Housing.csv")

# Map 'yes' and 'no' to boolean values
boolean_columns = ['mainroad', 'guestroom', 'basement', 'hotwaterheating', 'airconditioning', 'prefarea']
House_Data[boolean_columns] = House_Data[boolean_columns].replace({'yes': True, 'no': False})

# Display the first few rows of the cleaned DataFrame
hdt = House_Data
hdt.head()
#print(House_Data.head())


In [ ]:
import pandas as pd
import matplotlib.pyplot as plt

# Load the data from the CSV file
House_Data = pd.read_csv("/kaggle/input/housing-dataset/Housing.csv")

# Check for missing values
missing_values = House_Data.isnull().sum()
print("Missing Values:\n", missing_values)

# Drop rows with missing values
House_Data.dropna(inplace=True)

# Map 'yes' and 'no' to boolean values
boolean_columns = ['mainroad', 'guestroom', 'basement', 'hotwaterheating', 'airconditioning', 'prefarea']
House_Data[boolean_columns] = House_Data[boolean_columns].replace({'yes': True, 'no': False})

# Convert 'furnishingstatus' to numerical
House_Data['furnishingstatus'] = House_Data['furnishingstatus'].map({'furnished': 2, 'semi-furnished': 1, 'unfurnished': 0})

# Prepare data for plotting
plot_data = House_Data[['price', 'area', 'bedrooms', 'bathrooms', 'stories', 'parking', 'furnishingstatus']]

# Plot histograms for each column
plt.figure(figsize=(12, 8))
for i, col in enumerate(plot_data.columns):
    plt.subplot(3, 3, i+1)
    plt.hist(plot_data[col], bins=20, color='skyblu', edgecolor='black')
    plt.xlabel(col)
    plt.ylabel('Frequency')
    plt.tight_layout()

plt.show()
